src/sql/workbench/contrib/notebook/test/browser/notebookContexts.test.ts

```diff
@@ -0,0 +1,129 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as TypeMoq from 'typemoq';
+import * as assert from 'assert';
+import { nb } from 'azdata';
+import { NotebookContexts } from 'sql/workbench/contrib/notebook/browser/models/notebookContexts';
+import { IConnectionManagementService } from 'sql/platform/connection/common/connectionManagement';
+import { TestConnectionManagementService } from 'sql/platform/connection/test/common/testConnectionManagementService';
+import { ConnectionProfile } from 'sql/platform/connection/common/connectionProfile';
+import { TestCapabilitiesService } from 'sql/platform/capabilities/test/common/testCapabilitiesService';
+import { mssqlProviderName } from 'sql/platform/connection/common/constants';
+import { IDefaultConnection } from 'sql/workbench/contrib/notebook/browser/models/modelInterfaces';
+
+suite('Notebook Contexts', function (): void {
+	const defaultContext = NotebookContexts.DefaultContext;
+	const localContext = NotebookContexts.LocalContext;
+
+	function createTestConnProfile(): ConnectionProfile {
+		return new ConnectionProfile(new TestCapabilitiesService(), {
+			connectionName: 'Test',
+			savePassword: false,
+			groupFullName: 'testGroup',
+			serverName: 'testServerName',
+			databaseName: 'testDatabaseName',
+			authenticationType: 'integrated',
+			password: 'test',
+			userName: 'testUsername',
+			groupId: undefined,
+			providerName: mssqlProviderName,
+			options: {},
+			saveProfile: true,
+			id: 'testId'
+		});
+	}
+
+	test('Get Contexts For Kernel', async function (): Promise<void> {
+		const connService: TypeMoq.Mock<IConnectionManagementService>
+			= TypeMoq.Mock.ofType<IConnectionManagementService>(TestConnectionManagementService, TypeMoq.MockBehavior.Strict);
+
+		// No kernel or profile info provided
+		let conns = NotebookContexts.getContextsForKernel(connService.object, [mssqlProviderName]);
+		assert.deepEqual(conns, defaultContext);
+
```

Comment:
It's generally a good idea to add an assert message here when it fails - the default message is pretty useless for determining what actually went wrong.

Reply:
Looks like the asserts don't print the object contents when a message is provided for these.

Reply:
Does it do that by default? I forget the exact format the default message has. But you can always print them yourself using JSON.stringify if that's the case. It's just that line numbers aren't always easy to map back to the exact line that failed so when you have similar asserts in a row like you do it can make it annoying to figure out exactly which one failed.

Reply:
I just added the message argument, and that was the failure behavior. Since we print out the full results here on failure I'm content to leave it as is. Messages are generally way more helpful when dealing with things like generic assertTrue test cases.

Reply:
There's even a red/green diff message when using the default behavior.